app/services/query.py

```diff
@@ -0,0 +1,326 @@
+        except Host.DoesNotExist:
+            return None
+
+    def put(self, host):
+        self._dict_to_pynamo_host(host).save()
+
+    def batch_put(self, hosts):
+        '''
+        Note! Batched writes in pynamo are NOT ATOMIC. Batch writes are
+        done in groups of 25 with pynamo handling retries for partial failures
+        in a batch. It's possible that retries can be exhausted and we could
+        end up in a state where some weights were written and others weren't,
+        so external users should always ensure that weights were all
+        propagated and explicitly retry if not.
+
+        It's also possible that we're overriding newer data from hosts since
+        we're putting the whole host object rather than just updating an
+        individual field. This should be OK in practice as the time frame is
+        short and the next host update will return things to normal.
+        '''
+
+        # TODO need to look at the exceptions dynamo can throw here, catch, return False
+        with Host.batch_write() as batch:
+            for host in hosts:
+                batch.save(self._dict_to_pynamo_host(host))
+        return True
+
+    # TODO is all of this pomp and circumstance really necessary to properly delete?
+    #      need to better understand dynamodb
+    def delete(self, service, ip_address):
+        statsd = get_stats('service.host')
+        hosts = list(self._read_cursor(Host.query(service, ip_address__eq=ip_address)))
+        if len(hosts) == 0:
+            logging.error(
+                "Delete called for nonexistent host: service=%s ip=%s" % (service, ip_address)
+            )
+            return False
+        elif len(hosts) > 1:
+            logging.error(
+                "Returned more than 1 result for query %s %s.  Aborting the delete"
+                % (service, ip_address)
+            )
+            return False
+        else:
+            self._dict_to_pynamo_host(hosts[0]).delete()
+            statsd.incr("delete.%s" % service)
+            return True
+
+    def _read_cursor(self, cursor):
+        '''Converts a pynamo cursor into a generator.
+
+        :param cursor: pynamo cursor
+
+        :type cursor: TODO dig it up, some pynamo nonsense
+
+        :returns: generator based on the cursor
+        :retype: generator(dict)
+        '''
+        for host in cursor:
+            yield self._pynamo_host_to_dict(host)
+
+    def _pynamo_host_to_dict(self, host):
+        '''Converts a pynamo host into a dict.
+
+        :param host: pynamo host
+
+        :type host: Host
+
+        :returns: dictionary with host info
+        :rtype: dict
+        '''
+        _host = {}
+        _host['service'] = host.service
+        _host['ip_address'] = host.ip_address
+        _host['service_repo_name'] = host.service_repo_name
+        _host['port'] = host.port
+        _host['revision'] = host.revision
+        _host['last_check_in'] = host.last_check_in
```

Comment:
I was running tests on private, we have only one failures :) hooray,
```TypeError: datetime.datetime(2016, 10, 28, 20, 57, 9, 811399, tzinfo=<StaticTzInfo 'Etc/UTC'>) is not JSON serializable```
you'll have to revert str() conversion here, as it was in https://github.com/lyft/discovery/pull/4/files#diff-85c273d6ec3fda858eaa2cbf6619ab3bL233

Let's make tests passing, after that will make final pass and we can merge thsi